embassy-stm32/src/usart/mod.rs

```diff
@@ -55,6 +56,8 @@ pub struct Config {
     /// read will abort, the error reported and cleared
     /// if false, the error is ignored and cleared
     pub detect_previous_overrun: bool,
+    // None: automatically determine best oversampling ratio
+    pub force_oversampling: Option<u8>,
```

Comment:
I think this would be more idiomatic as an enum, with cases for "auto, 8, 16". This way you can't set invalid values at all.

Reply:
Done.